wp-admin/js/widgets/media-widgets.js

```diff
@@ -713,6 +713,20 @@ wp.mediaWidgets = ( function( $ ) {
 				modelProps.attachment_id = mediaFrameProps.id;
 			}
 
+			isEmbed = 0 === modelProps.attachment_id;
+
+			// Reset all video/audio sources.
+			wp.media.view.settings.embedExts.map( function( ext ) {
+
+				// Only reset when the main source changes.
+				if ( ! ( ext in modelProps ) || isEmbed ) {
+					modelProps[ ext ] = '';
+				}
+			} );
+
+			extension = mediaFrameProps.url.split( '.' ).pop().toLowerCase();
+			modelProps[ extension ] = mediaFrameProps.url;
```

Comment:
I'm more interested in the case where `mediaFrameProps.url` is equal to `https://www.youtube.com/watch?v=HTgYHHKs0Zw`, as in that case the “extension” will be `com/watch?v=HTgYHHKs0Zw`. So yeah, I guess we should skip this entirely if it doesn't have a file extension. Also, I think we should perhaps explicitly strip off any query or hash fragments from the URL so that if someone supplies a file URL with any additional params, the extension will get picked up properly.

In other words, something like:

```js
extension = mediaFrameProps.url.replace( /#.*$/, '' ).replace( /\?.*$/, '' ).split( '.' ).pop().toLowerCase()
```

And then yes, I think it would probably be good form to check if `extension` exists among `control.model.schema`.